tests/unit_tests/cuda/utils_ranges_cuda_kernel.cu

```diff
@@ -0,0 +1,185 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include "detray/definitions/cuda_definitions.hpp"
+#include "utils_ranges_cuda_kernel.hpp"
+
+namespace detray {
+
+//
+// single
+//
+__global__ void single_kernel(const dindex value, dindex* result) {
+
+    // single view should ony add the value 'i' once
+    for (auto i : detray::views::single(value)) {
+        *result += i;
+    }
+}
+
+void test_single(const dindex value, dindex& check) {
+    dindex* result{nullptr};
+    cudaMallocManaged(&result, sizeof(dindex));
```

Comment:
Maybe you can transfer the single value with vecmem pointer but it's trivial

Reply:
Ok, I did not know that existed :D

Reply:
Could you maybe 'point' me there?

Reply:
Hmm I tried to write a test kernel to check this out, but vecmem unique_ptr header is not compiled in device code in my PC :/... But I have seen codes that call unique_ptr in the device (acts-project/traccc#230) so there might be something wrong in my environment setup. Anyway it's not important so we can go with this

Reply:
I am not sure a unique pointer makes much sense for a single index? Or is it not meant to do memory management in vecmem?